### PR TITLE
feat: [PLG-3003]: Added AB test for work emails

### PR DESCRIPTION
**Pull request by devinder-harness**
This PR adds an AB test to auth ui which blocks signup for free domains like 
* "gmail.com",
* "yahoo.com",
* "hotmail.com",
* "outlook.com",
* "qq.com"

### CODE REVIEW

**Comment by devinder-harness** [comment]
retrigger jest

File: src/components/Text/Text.tsx [hunk 1]
```diff
@@ -15,16 +15,18 @@ interface TextProps {
...
   children,
   className,
   icon,
-  iconProps
+  iconProps,
+  hasError
 }) => {
   return (
-    <span className={cx(css.text, className)}>
+    <span className={cx(css.text, className, { [css.erorText]: hasError })}>
```
Comment (abhinavrastogi-harness) [inline src/components/Text/Text.tsx:29]: spelling.. `css.errorText`

File: src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx [hunk 2]
```diff
@@ -28,7 +28,12 @@ import { CATEGORY, PAGE, EVENT, FF_MAP } from "utils/TelemetryUtils";
 import SignupFormWithCredentials from "./SignupFormWithCredentials";
 import SignupFormOAuth from "./SignupFormOAuth";
 import BasicLayoutExperimental from "./BasicLayout/BasicLayoutExperimental";
-import { getModuleDetails, updateReferer } from "./utils";
+import {
+  EMAIL_RESTRICITION_FLAG_VARIANTS,
```
Comment (abhinavrastogi-harness) [inline src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx:32]: spelling

File: src/pages/SignUp/SaasExperimentalForms/SignupFormWithCredentials.tsx [hunk 3]
```diff
@@ -70,28 +73,33 @@ const SignupFormWithCredentials = ({
...
+              utm_medium: utm_medium || "",
+              utm_campaign: utm_campaign || "",
+              utm_term: utm_term || "",
+              utm_content: utm_content || ""
+            }
+          });
+        }}
+      />
+      {hasRestrictedEmail && (
+        <Text hasError>Please enter your work email.</Text>
```
Comment (sahithikolichala) [inline src/pages/SignUp/SaasExperimentalForms/SignupFormWithCredentials.tsx:100]: strings file

File: src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx [hunk 4]
```diff
@@ -164,6 +172,30 @@ const SignUpExperimental: React.FC = () => {
...
+    if (
+      EmailRestrictionSetting ===
+      EMAIL_RESTRICTION_FLAG_VARIANTS.RESTRICTION_ENABLED
+    ) {
+      if (isRestrictedEmail(data.email)) {
+        setRestrictedEmail(true);
+        telemetry.track({
+          event: EVENT.SIGNUP_FREE_DOMAIN_ERROR,
+          properties: {
+            intent: module || "",
```
Comment (sahithikolichala) [inline src/pages/SignUp/SaasExperimentalForms/SignUpExperimental.tsx:184]: this code seems to be redundant, can we make it commo and use

File: src/pages/SignUp/SaasExperimentalForms/SignupFormWithCredentials.tsx (hunk 3 above)
Comment (devinder-harness) [reply src/pages/SignUp/SaasExperimentalForms/SignupFormWithCredentials.tsx:100]: no string file in auth ui

File: src/pages/SignUp/SaasExperimentalForms/SignupFormWithCredentials.tsx [hunk 5]
```diff
@@ -70,28 +73,33 @@ const SignupFormWithCredentials = ({
...
+        placeholder="email@work.com"
+        disabled={loading || captchaExecuting}
+        validate={validateEmail}
+        autoFocus
+        onBlur={(e: FocusEvent<HTMLInputElement>) => {
+          telemetry.track({
+            event: EVENT.EMAIL_INPUT,
+            properties: {
+              category: CATEGORY.SIGNUP,
+              email: e.target.value,
```
Comment (sahithikolichala) [inline src/pages/SignUp/SaasExperimentalForms/SignupFormWithCredentials.tsx:89]: can we not use utmProps here

**Review by sahithikolichala — APPROVED** [review]

**Pull request merged**